build-clang-6.0.0.sh

```diff
@@ -0,0 +1,160 @@
+#!/bin/bash
```

Comment:
It would be nice to re-use the same script for 5.0.1 and 6.0.0, given the differences seem to be:
1. Actual version numbers
2. A 'final' being present or absent in a string
3. Extracting two extra directories

Reply:
I agree, just wasn't sure if that was something people would find useful. This works for clang 5.0.1 and 6.0.0 now using e.g. `./build-clang.sh osx 5.0.1` (or it worked when I last tried running it, will focus on correctness later today).